crates/iceberg/src/spec/schema/mod.rs

```diff
@@ -127,6 +127,22 @@ impl SchemaBuilder {
 
     /// Builds the schema.
     pub fn build(self) -> Result<Schema> {
+        // If field IDs need to be reassigned, do it first before validation
+        if let Some(start_from) = self.reassign_field_ids_from {
```

Comment:
What's the motivation of this change? The problem with this change is that it may make the error more difficult to read. For example, if user passed a non exists identifier id, originally the error message would be quite clear and easy to understand. But this change will confuse user.

Reply:
This is a good point!

My original intention was to avoid building id-to-field map again when we are trying to reassign field ids.